packages/ra-keycloak/src/authProvider.ts

```diff
@@ -188,27 +187,24 @@ export const keycloakAuthProvider = (
  * It seems the Keycloak init function may initially return before having authenticated the user.
  * To ensure we have the correct state, we need to wait for the onAuthSuccess event.
  */
-const isAuthenticated = (keycloakClient: Keycloak, timeout = 500) => {
+const isAuthenticated = (keycloakClient: Keycloak, timeout = 1000) => {
```

Comment:
This was an important change too, on my computer, keycloak-js took around 500ms to process the token, so each second try it would fail in timeout although the user is actually signed in alright.
Increasing this timeout to 1000ms fixed it.

I also tried to additionally implement `onAuthError` (to avoid waiting for the full timeout when not logged in), but with no luck.

1000ms + the ability to change this timeout via a prop is enough IMO.

Additional notes: some timing measurements on my computer to back my words up:

```
isAuthenticated: 488.92919921875 ms
isAuthenticated: 511.9560546875 ms
isAuthenticated: 575.23291015625 ms
```